<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Perform unsupervised clustering to capture population structure based on iterative pruning — ipcaps • IPCAPS</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Perform unsupervised clustering to capture population structure based on iterative pruning — ipcaps" />
<meta property="og:description" content="This version supports ordinal data which can be applied directly
to SNP data to identify fine-scale population structure. It was built on the
iterative pruning Principal Component Analysis (ipPCA) algorithm
(Intarapanich et al., 2009; Limpiti et al., 2011). The IPCAPS involves an
iterative process using multiple splits based on multivariate Gaussian
mixture modeling of principal components and Clustering EM estimation (Lebret
et al., 2015). In each iteration, rough clusters and outliers are also
identified using our own method called rubikclust (R package KRIS)." />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">IPCAPS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/kridsadakorn/ipcaps/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perform unsupervised clustering to capture population structure based on iterative pruning</h1>
    <small class="dont-index">Source: <a href='https://github.com/kridsadakorn/ipcaps/blob/master/R/ipcaps.R'><code>R/ipcaps.R</code></a></small>
    <div class="hidden name"><code>ipcaps.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This version supports ordinal data which can be applied directly
to SNP data to identify fine-scale population structure. It was built on the
iterative pruning Principal Component Analysis (ipPCA) algorithm
(Intarapanich et al., 2009; Limpiti et al., 2011). The IPCAPS involves an
iterative process using multiple splits based on multivariate Gaussian
mixture modeling of principal components and Clustering EM estimation (Lebret
et al., 2015). In each iteration, rough clusters and outliers are also
identified using our own method called rubikclust (R package <span class="pkg">KRIS</span>).</p>
    </div>

    <pre class="usage"><span class='fu'>ipcaps</span><span class='op'>(</span>
  bed <span class='op'>=</span> <span class='cn'>NA</span>,
  rdata <span class='op'>=</span> <span class='cn'>NA</span>,
  files <span class='op'>=</span> <span class='cn'>NA</span>,
  label.file <span class='op'>=</span> <span class='cn'>NA</span>,
  lab.col <span class='op'>=</span> <span class='fl'>1</span>,
  <span class='va'>out</span>,
  plot.as.pdf <span class='op'>=</span> <span class='cn'>FALSE</span>,
  method <span class='op'>=</span> <span class='st'>"mix"</span>,
  missing <span class='op'>=</span> <span class='cn'>NA</span>,
  covariate <span class='op'>=</span> <span class='cn'>NA</span>,
  cov.col.first <span class='op'>=</span> <span class='cn'>NA</span>,
  cov.col.last <span class='op'>=</span> <span class='cn'>NA</span>,
  threshold <span class='op'>=</span> <span class='fl'>0.18</span>,
  min.fst <span class='op'>=</span> <span class='fl'>8e-04</span>,
  min.in.group <span class='op'>=</span> <span class='fl'>20</span>,
  no.plot <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>bed</th>
      <td><p>A PLINK binary format consists of 3 files; bed, bim, and fam. To
generate these files from PLINK, use option –make-bed. See more details at:
<a href='http://zzz.bwh.harvard.edu/plink/data.shtml'>harvard.edu</a>.</p></td>
    </tr>
    <tr>
      <th>rdata</th>
      <td><p>In case of re-analysis, it is convenient to run IPCAPS using the
file rawdata.RData generated by IPCAPS. This file contains a matrix of SNPs
(raw.data) and a vector of labels (label).</p></td>
    </tr>
    <tr>
      <th>files</th>
      <td><p>IPCAPS supports SNPs encoded as 0, 1 and 2 (dosage encoding).
Rows represent SNPs and columns represent subjects. Each column needs to be
separated by a space or a tab. A big text file should be divided into smaller
files to load faster. For instance, to input 3 files, use as: files=c(
'input1.txt', 'input2.txt', 'input3.txt').</p></td>
    </tr>
    <tr>
      <th>label.file</th>
      <td><p>An additional useful information (called "labels" in
IPCAPS) related subject, for example, geographic location or disease
phenotype. These labels (one at a time) are used in displaying the clustering
outcome of IPCAPS. A label file must contain at least one column. However,
it may contain more than one column in which case each column need to be
separated by a space or a tab.</p></td>
    </tr>
    <tr>
      <th>lab.col</th>
      <td><p>The label in the label file to be used in the tree-like
display of IPCAPS clustering results.</p></td>
    </tr>
    <tr>
      <th>out</th>
      <td><p>To set an absolute path for IPCAPS output. If the specified output
directory already exists, result files are saved in sub-directories
cluster_out, cluster_out1, cluster_out2, etc.</p></td>
    </tr>
    <tr>
      <th>plot.as.pdf</th>
      <td><p>To export plots as PDF. When omitted, plots are saved as PNG.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>The internal clustering method. It can be set to "mix"
(rubikclust &amp; mixmod), "mixmod" (Lebret et al., 2015), "clara" (R: Clustering
Large Applications), "pam" (R: Partitioning Around Medoids (PAM) Object),
"meanshift" (Wang, 2016), "apcluster" (Bodenhofer et al., 2016), and "hclust"
(R: Hierarchical Clustering). Default = "mix".</p></td>
    </tr>
    <tr>
      <th>missing</th>
      <td><p>Symbol used for missing genotypes. Default = NA.</p></td>
    </tr>
    <tr>
      <th>covariate</th>
      <td><p>A file of covariates; one covariate per column. SNPs can be
adjusted for these covariates via regression modeling and residual
computation.</p></td>
    </tr>
    <tr>
      <th>cov.col.first</th>
      <td><p>Refer to a covariate file, the first covariate to be
considered as confounding variable.</p></td>
    </tr>
    <tr>
      <th>cov.col.last</th>
      <td><p>Refer to a covariate file, the last covariate to be
considered as confounding variable. All the variables in between the
cov.col.first and cov.col.last will be considered in the adjustment process.</p></td>
    </tr>
    <tr>
      <th>threshold</th>
      <td><p>Cutoff value for EigenFit. Possible values range from 0.03
to 0.18. The smaller, the potentially finer the substructure can be. Default
= 0.18.</p></td>
    </tr>
    <tr>
      <th>min.fst</th>
      <td><p>Minimum Fst between a pair of subgroups. Default = 0.0008.</p></td>
    </tr>
    <tr>
      <th>min.in.group</th>
      <td><p>Minimum number of individuals to constitute a cluster or
subgroup. Default = 20.</p></td>
    </tr>
    <tr>
      <th>no.plot</th>
      <td><p>No plot is generated if this option is TRUE. This option is
useful when the system does not support X Windows in the unix based system.
Default = FALSE.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Returns the list object containing 2 internal objects;
output.dir as class character and cluster as class data.frame. The object
output.dir stores a result directory. The object cluster contains 4 columns,
group, node, label, and row.number. The column group contains the assigned
groups from IPCAPS. The column node contains node numbers in a tree as
illustrated in the HTML result files. The column label contains the given
labels that is set to the parameter label. The column row.number contains
indices to an input data, which is matched to row numbers of input matrix.
All clustering result files are saved in one directory (as specified by out)
containing assigned groups, hierarchical trees of group membership, PC plots,
and binary data for further analysis.</p><ul>
<li><p><code>$snp</code> is a SNP matrix from BED file.</p></li>
<li><p><code>$snp.info</code> is a data.frame of SNP information from BIM file.</p></li>
<li><p><code>$ind.info</code> is a data.frame of individual information from FAM file.</p></li>
</ul><p>If function return <code>NULL</code>, it means the input files are not in proper
format.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The computational time depends on the number of individuals.
Consequentially, if large data sets are analyzed, it may be necessary first
to split data into smaller files, and then load into computer memory (with
parameter 'files'). Internally, the split files are merged to construct
a com-prehensive matrix.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Bodenhofer, U., Palme, J., Melkonian, C., and Kothmeier, A. (2016). apcluster
: Affinity Propagation Clustering. Available at <a href=' https://CRAN.R-project.org/package=apcluster'>CRAN</a> (Accessed March 7, 2017).</p>
<p>Intarapanich, A., Shaw, P. J., Assawamakin, A., Wangkumhang, P., Ngamphiw, C.
, Chaichoompu, K., et al. (2009). Iterative pruning PCA improves resolution
of highly structured populations. BMC Bioinformatics 10, 382. doi:10.1186/
1471-2105-10-382.</p>
<p>Lebret, R., Iovleff, S., Langrognet, F., Biernacki, C., Celeux, G., and
Govaert, G. (2015). Rmixmod: TheRPackage of the Model-Based Unsupervised,
Supervised, and Semi-Supervised ClassificationMixmodLibrary. J. Stat. Softw.
67. doi:10.18637/jss.v067.i06.</p>
<p>Limpiti, T., Intarapanich, A., Assawamakin, A., Shaw, P. J., Wangkumhang, P.,
Piriyapongsa, J., et al. (2011). Study of large and highly stratified
population datasets by combining iterative pruning principal component
analysis and structure. BMC Bioinformatics 12, 255. doi:10.1186/1471-2105-12-
255.</p>
<p>Maechler, M., Rousseeuw, P., Struyf, A., Hubert, M., and Hornik, K. (2017).
cluster: Cluster Analysis Basics and Extensions.</p>
<p>R: Clustering Large Applications Available at: <a href='https://stat.ethz.ch/R-manual/R-devel/library/cluster/html/clara.html'>ethz.ch</a> (Accessed March 7, 2017).</p>
<p>R Core Team (2017). R: A Language and Environment for Statistical Computing.
Vienna, Austria: R Foundation for Statistical Computing Available at:
<a href='https://www.R-project.org/'>r-project.org</a>.</p>
<p>R: Hierarchical Clustering Available at: <a href='https://stat.ethz.ch/R-manual/R-devel/library/stats/html/hclust.html'>ethz.ch</a> (Accessed March 7, 2017).</p>
<p>R: Partitioning Around Medoids (PAM) Object Available at: <a href='https://stat.ethz.ch/R-manual/R-devel/library/cluster/html/pam.object.html'>ethz.ch</a> (Accessed
March 7, 2017).</p>
<p>Wang, M. C. and D. (2016). MeanShift: Clustering via the Mean Shift
Algorithm. Available at: <a href='https://CRAN.R-project.org/package=MeanShift'>CRAN</a>
(Accessed March 7, 2017).</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>#Use the BED format as input</span>
<span class='co'>#Importantly, bed file, bim file, and fam file are required</span>
<span class='co'>#Use the example files embedded in the package</span>

<span class='va'>BED.file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"ipcaps_example.bed"</span>, package <span class='op'>=</span> <span class='st'>"IPCAPS"</span><span class='op'>)</span>
<span class='va'>LABEL.file</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"ipcaps_example_individuals.txt.gz"</span>,
                          package <span class='op'>=</span> <span class='st'>"IPCAPS"</span><span class='op'>)</span>
<span class='va'>my.cluster1</span> <span class='op'>&lt;-</span> <span class='fu'>ipcaps</span><span class='op'>(</span>bed <span class='op'>=</span> <span class='va'>BED.file</span>, label.file <span class='op'>=</span> <span class='va'>LABEL.file</span>, lab.col <span class='op'>=</span> <span class='fl'>2</span>,
out <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; Running ... IPCAPS 
#&gt; 	output: /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm 
#&gt; 	label file: /private/var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T/RtmpRPmTOI/temp_libpath1030f226cd8b6/IPCAPS/extdata/ipcaps_example_individuals.txt.gz
#&gt; 	label column: 2
#&gt; 	threshold: 0.18
#&gt; 	minimum Fst: 8e-04
#&gt; 	maximum thread: 1
#&gt; 	method: mix
#&gt; 	bed: /private/var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T/RtmpRPmTOI/temp_libpath1030f226cd8b6/IPCAPS/extdata/ipcaps_example.bed
#&gt; 	minimum in group: 20
#&gt; 	data type: snp
#&gt; 	missing: NA
#&gt; In preprocessing step
#&gt; Note: the directory '/var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm' is existed. 
#&gt; The result files will be saved to this directory: /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2 
#&gt; Input data: 103 individuals, 2000 markers
#&gt; Start calculating
#&gt; Node 1: Start the process
#&gt; Node 1: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 1: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/condition.RData
#&gt; Node 1: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/rawdata.RData
#&gt; Node 1: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node1.RData
#&gt; Node 1: Check for splitting
#&gt; Node 1: Reducing matrix
#&gt; Node 1: EigenFit = 0.652486925995028, Threshold = 0.18, no. significant PCs = 3
#&gt; Node 1: Start clustering
#&gt; Node 1: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node1.RData
#&gt; Node 1: done for clustering
#&gt; Time difference of 0.1008899 secs
#&gt; Node 1: Checking Fst of group 1 and group 2
#&gt; Node 1: Checking Fst of group 1 and group 3
#&gt; Node 1: Checking Fst of group 1 and group 4
#&gt; Node 1: Checking Fst of group 2 and group 3
#&gt; Node 1: Checking Fst of group 2 and group 4
#&gt; Node 1: Checking Fst of group 3 and group 4
#&gt; Node 1 times took 0.28612494468689 secs
#&gt; Node 1: 103 individuals were splitted into: 4 groups
#&gt; Node 1: Return status 0
#&gt; Node 1: Split to sub-nodes
#&gt; Node 1: Saving /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node2.RData
#&gt; Node 1: Saving /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node3.RData
#&gt; Node 1: Saving /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node4.RData
#&gt; Node 1: Saving /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node5.RData
#&gt; Node 1: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 1: Updating /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 1: Done!
#&gt; Time difference of 0.5675111 secs
#&gt; Node 2: Start the process
#&gt; Node 2: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 2: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/condition.RData
#&gt; Node 2: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/rawdata.RData
#&gt; Node 2: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node2.RData
#&gt; Node 2: Check for splitting
#&gt; Node 2: A number of node is lower than the minimum number (20), therefore split was not performed
#&gt; Node 2: Return status 1
#&gt; Node 2: No split was performed, Status=1
#&gt; Node 2: Updating /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/leafnode.RData
#&gt; Node 2: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 2: Updating /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 2: Done!
#&gt; Time difference of 0.02910995 secs
#&gt; Node 3: Start the process
#&gt; Node 3: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 3: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/condition.RData
#&gt; Node 3: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/rawdata.RData
#&gt; Node 3: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node3.RData
#&gt; Node 3: Check for splitting
#&gt; Node 3: A number of node is lower than the minimum number (20), therefore split was not performed
#&gt; Node 3: Return status 1
#&gt; Node 3: No split was performed, Status=1
#&gt; Node 3: Updating /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/leafnode.RData
#&gt; Node 3: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 3: Updating /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 3: Done!
#&gt; Time difference of 0.03018904 secs
#&gt; Node 4: Start the process
#&gt; Node 4: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 4: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/condition.RData
#&gt; Node 4: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/rawdata.RData
#&gt; Node 4: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node4.RData
#&gt; Node 4: Check for splitting
#&gt; Node 4: A number of node is lower than the minimum number (20), therefore split was not performed
#&gt; Node 4: Return status 1
#&gt; Node 4: No split was performed, Status=1
#&gt; Node 4: Updating /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/leafnode.RData
#&gt; Node 4: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 4: Updating /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 4: Done!
#&gt; Time difference of 0.02878499 secs
#&gt; Node 5: Start the process
#&gt; Node 5: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 5: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/condition.RData
#&gt; Node 5: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/rawdata.RData
#&gt; Node 5: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node5.RData
#&gt; Node 5: Check for splitting
#&gt; Node 5: Reducing matrix
#&gt; Node 5: EigenFit = 0.343228072167861, Threshold = 0.18, no. significant PCs = 3
#&gt; Node 5: Start clustering
#&gt; Node 5: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node5.RData
#&gt; Node 5: done for clustering
#&gt; Time difference of 0.0922451 secs
#&gt; Node 5: Checking Fst of group 1 and group 2
#&gt; Node 5 times took 0.137973785400391 secs
#&gt; Node 5: 100 individuals were splitted into: 2 groups
#&gt; Node 5: Return status 0
#&gt; Node 5: Split to sub-nodes
#&gt; Node 5: Saving /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node6.RData
#&gt; Node 5: Saving /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node7.RData
#&gt; Node 5: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 5: Updating /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 5: Done!
#&gt; Time difference of 0.2850099 secs
#&gt; Node 6: Start the process
#&gt; Node 6: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 6: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/condition.RData
#&gt; Node 6: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/rawdata.RData
#&gt; Node 6: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node6.RData
#&gt; Node 6: Check for splitting
#&gt; Node 6: Reducing matrix
#&gt; Node 6: EigenFit = 0.0367738545252401, Threshold = 0.18, no. significant PCs = 
#&gt; Node 6: No split was performed because EigenFit is lower than threshold
#&gt; Node 6: Return status 1
#&gt; Node 6: No split was performed, Status=1
#&gt; Node 6: Updating /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/leafnode.RData
#&gt; Node 6: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 6: Updating /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 6: Done!
#&gt; Time difference of 0.09509611 secs
#&gt; Node 7: Start the process
#&gt; Node 7: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 7: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/condition.RData
#&gt; Node 7: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/rawdata.RData
#&gt; Node 7: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/node7.RData
#&gt; Node 7: Check for splitting
#&gt; Node 7: Reducing matrix
#&gt; Node 7: EigenFit = 0.0221170820186458, Threshold = 0.18, no. significant PCs = 
#&gt; Node 7: No split was performed because EigenFit is lower than threshold
#&gt; Node 7: Return status 1
#&gt; Node 7: No split was performed, Status=1
#&gt; Node 7: Updating /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/leafnode.RData
#&gt; Node 7: Loading /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 7: Updating /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/RData/tree.RData
#&gt; Node 7: Done!
#&gt; Time difference of 0.1078601 secs
#&gt; In post process step
#&gt; Exporting node 2 as group 1
#&gt; Exporting node 3 as group 2
#&gt; Exporting node 4 as group 3
#&gt; Exporting node 6 as group 4
#&gt; Exporting node 7 as group 5
#&gt; Note: save as /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2/groups.txt</div><div class='output co'>#&gt; The result files were saved at:  /var/folders/sp/hhmj9xvx53z4g4dktf5f503r0000gp/T//RtmpYpZQgm/cluster_output2 
#&gt; Total runtime is 2.10697603225708 sec</div><div class='input'>
<span class='fu'><a href='https://rdrr.io/r/base/table.html'>table</a></span><span class='op'>(</span><span class='va'>my.cluster1</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>$</span><span class='va'>label</span>, <span class='va'>my.cluster1</span><span class='op'>$</span><span class='va'>cluster</span><span class='op'>$</span><span class='va'>group</span><span class='op'>)</span>
</div><div class='output co'>#&gt;           
#&gt;             1  2  3  4  5
#&gt;   outlier3  1  1  1  0  0
#&gt;   pop1      0  0  0  0 50
#&gt;   pop2      0  0  0 50  0</div><div class='input'>
<span class='co'># Alternatively, use a text file as input</span>
<span class='co'># Use the example files embedded in the package</span>

<span class='co'>#text.file &lt;- system.file("extdata", "ipcaps_example_rowVar_colInd.txt.gz",</span>
<span class='co'>#                          package="IPCAPS")</span>
<span class='co'>#LABEL.file &lt;- system.file("extdata", "ipcaps_example_individuals.txt.gz",</span>
<span class='co'>#                           package="IPCAPS")</span>

<span class='co'>#my.cluster2 &lt;- ipcaps(files = c(text.file), label.file = LABEL.file, lab.col = 2,</span>
<span class='co'>#                       out=tempdir())</span>
<span class='co'>#table(my.cluster2$cluster$label, my.cluster2$cluster$group)</span>

<span class='co'># The other alternative way, use an R Data file as input</span>
<span class='co'># Use the example file embedded in the package</span>

<span class='co'>#rdata.file &lt;- system.file("extdata", "ipcaps_example.rda", package = "IPCAPS")</span>

<span class='co'>#my.cluster3 &lt;- ipcaps(rdata = rdata.file, out = tempdir())</span>
<span class='co'>#table(my.cluster3$cluster$label, my.cluster3$cluster$group)</span>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Kridsadakorn Chaichoompu, Kristel Van Steen, Fentaw Abegaz, Sissades Tongsima, Philip James Shaw, Anavaj Sakuntabhai, Luisa Pereira.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


